index.html

```diff
@@ -0,0 +1,201 @@
+}
+```
+
+### Requiring `@using` within a spec
+
+Specifications **may** include the following language to instruct schema authors to use `@using` to indicate documents which are intended to conform to the specification:
+
+> SPEC_NAME schema documents **must**/**should**/**may** have a `schema` declaration with a [`@using` directive](/specs/use)
+> and the specification URL `SPEC_URL`:
+>
+> ```graphql
+> schema @using(spec: SPEC_URL) {
+>   query: Query
+> }
+> ```
+
+`SPEC_NAME` and `SPEC_URL` should be replaced with the names and URLs for the specification. For example:
+
+> Example Spec schema documents **must** have a `schema` declaration with a [`@using` directive](/specs/use)
+> and the specification URL `https://specs.example.com/exampleSpec/1`:
+>
+> ```graphql
+> schema @using(spec: "https://specs.example.com/exampleSpec/1") {
+>   query: Query
+> }
+> ```
+
+## Processing Schemas
+
+```mermaid
+#::[Example processing pipeline]
+graph LR
+  schema(["📄  Input Schema"]):::file-->proc("🤖 &nbsp;Processor")
+  proc-->output(["📄  Output Schema"]):::file
+  classDef file fill:none,color:#eee;
+  style proc fill:none,stroke:fuchsia,color:fuchsia;
+```
+
+A common case is that of a processor which consumes an input schema and generates an output schema.
+
+The general guidance for processor behavior is: don't react to what you don't understand.
+
+Specifically, processors:
+  - **should** pass through `@using` directives which reference unknown spec URLs
+  - **should** pass through unknown directives, types, and other schema elements
+  - **should** ignore meaningless URL components (such as unknown query parameters or fragments) within `@using` URLs
+
+An exception to this is processors which prepare the schema for final public consumption. Such processors **may** choose to eliminate all unknown directives and prefixed types in order to hide schema implementation details within the published schema. This will impair the operation of tooling which relies on these directives—such tools will not be able to run on the output schema, so the benefits and costs of this kind of information hiding should be weighed carefully on a case-by-case basis.
+
+## Spec URLs
+
+```graphql
+#::[Declaration of using_URL scalar]
+scalar using_URL
+  @specifiedBy(url: "https://apollographql.com/docs/specs/using#spec-urls")
+```
+
+The URL of a specification.
+
+Specification URLs **must** be valid [RFC 3986 URLs](https://tools.ietf.org/html/rfc3986). They **should** provide the content of the appropriate version of the specification in some human-readable form. It is not required that the content be machine-readable in any particular way.
+
+The final component of the URL's `"/"`-separated `path` **must** be a valid [SemVer specifier](#semver-specifiers) identifying the version of the spec in use. Everything prior to the version specifier is the spec's *base URL*. Processors **should** treat the spec's base URL as an opaque string identifying the specification.
```

Comment:
Continuation of [this comment](https://github.com/apollographql/specs/pull/1/files#r527908159).
>We did make our own, it's in the @using spec
>
>Specifically, the SemVer spec does provide ordering, we just describe the grammar of version and how to flesh them out into a semver for comparison
>
>It'll be simpler than what's provided by node-semver—mostly a prefix match

I'm fine with us having a simpler version of `node-semver` in the `@using` spec.

The SemVer spec does indeed provide a spec for ordering, although I'd like it if there were a bit more description around the constraints imposed by SemVer specifiers in the `@using` spec. (E.g. if it's just prefix matching, then just a statement of that.)

As an example, consider SemVer specifier `1.2`. If it's prefix matching, then the constraint would be `1.2.0 <= x < 1.3.0`. If we go by "version compatibility" (which is what's mentioned in the `@using` spec), then the constraint would be `1.2.0 <= x < 2.0.0`, since minor versions after `1.2.0` are required to be backwards compatible with `1.2.0`.

Reply:
There's also a bit of trickiness around prerelease precedence. For example when I use `< x.y.z` above, I mean to exclude prerelease versions of `x.y.z`, but `<` as defined by precedence rules would technically include them.  There's also a question of whether users would really want to select a prerelease version when using e.g. SemVer specifier `1.2`, even if that prerelease version has the highest precedence. (`node-semver` gets around this by saying that a prerelease version won't be considered unless that prerelease version has the same major/minor/patch version as a comparator string and that comparator also has a prerelease tag.)

Reply:
we should probably adopt that trick